CONTRIBUTING.md

```diff
@@ -0,0 +1,504 @@
+
+```bash
+cd aws-sdk-ios
+pod install
+xed .
+```
+
+Conversely if you wanted to make a change to our Auth SDK's the initial workflow might be:
+
+```bash
+cd aws-sdk-ios
+cd AWSAuthSDK
+pod install
+xed .
+```
+
+## Tools
+
+[Xcode](https://developer.apple.com/xcode/) and [cocoapods](https://guides.cocoapods.org/using/getting-started.html) are used for all build and dependency management.
+
+Some other widely used dependencies are:
+
+[Mantle](https://github.com/Mantle/Mantle) is what we use for deserialization and serialization and any modeling with JSON in our AWS iOS SDK.
+
+[SocketRocket](https://github.com/facebook/SocketRocket) is used for web socket connections.
+
+### Integration Testing Setup
+
+For our integration tests, which run on your development machine we use XCTest and an aws configuration file to set up to set up credentials. To set it up to build those integration tests on your machine, you will need to create a `credentials.json` file in inside AWSCoreTests> Resources>credentials.json and it should have the following format:
+
+```
+{
+    "accessKey": "XXXXXXXXXXXXXX",
+    "secretKey": "XXXXXXXXXXXXXXXXXXXXXXXXX",
+    "accountId": "XXXXXXXXXXXX",
+    "identityPoolId": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+    "unauthIdentityPoolId": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+    "authRoleArn": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX", 
+    "unauthRoleArn":"XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+    "amazon-connect-instance-id": "XXXXXXXXXXXXXXXXXXXXX",
+    "amazon-connect-destination-dial-in-phone-number": "XXXXXXXXXX",
+    "anotherAccountId": "XXXXXXXXXXXXX",
+    "apiEndpoint": "XXXXXXXXXXXXXXXXXXXXXXXXX",
+    "facebookAppId": "XXXXXXXXXXXX",
+    "facebookAppSecret": "XXXXXXXXXXXXXXXXXXXX",
+    "iot-custom-authorizer-name": "XXXXXXXXXX",
+    "iot-custom-authorizer-token-key-name": "XXXXXXXXXXXX",
+    "iot-custom-authorizer-token-signature": "XXXXXXXXXXXXXXXXXXXXXXXX",
+    "iot-custom-authorizer-token-value": "allow",
+    "iot-us-east1-endpoint": "XXXXXXXXXXXXXXXXXXXXXX",
+    "iot-us-east2-endpoint": "XXXXXXXXXXXXXXXXXXXXXX",
+    "pinpointAppId": "XXXXXXXXXXXXXXXXXXX",
+    "sageMaker-endpoint": "XXXXXXXXXXXXXXXX",
+    "snsTestAppArn": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+    "textract-s3-bucket-name": "XXXXXXXXXXXXXXXXXXXXX",
+    "textract-s3-object-name": "XXXXXXXXXXXXXXXXXXXXXX",
+    "verifiedEmailAddress": "XXXXXXXXXXXXXXX"
+}
+```
+
+This is the full file ( in case you would like to make future contributions later on ;) ) which would be quite a bit of AWS resource creation. However, for the purpose of running majority of our tests (as you can see some of the keys are very specific to the tests they are affecting aka. textract-s3-bucket-name is affecting textract integration tests) you probably just need to fill out the below keys and you should be able to run the integration tests just fine:
```

Comment:
It's unclear what you mean by "the full file".. can you explain, and maybe we can work on the wording for this one?

Reply:
like if you created every aws resource neeeded to run every integration test in our sdk. but no one is gonna do that. most people will make a small change to one of them and only need to run that set of integ tests

Reply:
adjusted wording